crates/metaslang/bindings/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+pub mod builder;
+
+use std::collections::{BTreeSet, HashMap};
+use std::fmt;
+use std::fmt::Debug;
+use std::iter::once;
+
+use metaslang_cst::cursor::Cursor;
+use metaslang_cst::KindTypes;
+use metaslang_graph_builder::ast::File;
+use metaslang_graph_builder::functions::Functions;
+use semver::Version;
+use stack_graphs::graph::StackGraph;
+use stack_graphs::partial::PartialPaths;
+use stack_graphs::stitching::{ForwardPartialPathStitcher, GraphEdgeCandidates, StitcherConfig};
+use thiserror::Error;
+
+type Builder<'a, KT> = builder::Builder<'a, KT>;
+type GraphHandle = stack_graphs::arena::Handle<stack_graphs::graph::Node>;
+
+#[derive(Error, Debug)]
+pub enum BindingsError {
+    #[error(transparent)]
+    BuildError(#[from] builder::BuildError),
+}
+
+pub struct Bindings<KT: KindTypes + 'static> {
+    version: Version,
+    graph_builder_file: File<KT>,
+    functions: Functions<KT>,
+    stack_graph: StackGraph,
+    cursors: HashMap<GraphHandle, Cursor<KT>>,
+}
+
+impl<KT: KindTypes + 'static> Bindings<KT> {
```

Comment:
I think this initial implementation would work well for us to make progress on the bindings definitions, but IIUC, the final version would expose such functionality through cursors instead, so that users can go from a current `Cursor` to see if it is bound to something (handle), and then use that handle to go to def/get refs/etc... Is that correct? Just TOL.

Reply:
Do you mean the `all_references` and `all_definitions` functions? Yes, although it may make sense to keep them. I think they may be useful. But anyway, yes, the main interface should be through cursors. It's not easy to do it yet, as we discussed on our last stand up. Cursors don't define ordering so it's not efficient to use them for indexing.